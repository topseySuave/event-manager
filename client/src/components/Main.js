import React, {Component} from 'react';
import { Route, Switch} from 'react-router-dom'
import HomePage from './homepage';
import SignUp from './authentication/signup/signup';
import SignIn from './authentication/signin/signin';
import CenterDetailIndex from './centerComponent/centerDetail';
import AllCentersIndex from './centerComponent/allCenters';
import NoMatch from './noMatch';
import createBrowserHistory from 'history/createBrowserHistory'

const history = createBrowserHistory({
    basename: '/'
});

class Main extends Component {
    render() {
        return (
            <Switch history={history}>
                <Route exact path ='/' component={HomePage} />
                <Route path ='/signup' component={SignUp} />
                <Route path = '/signin' component={SignIn} />
                <Route exact={false} path = '/center-detail/:identifier' component={CenterDetailIndex} />
                <Route path = '/centers' component={AllCentersIndex} />
                <Route path='*' component={NoMatch} />
            </Switch>
        )
    }
}

export default Main;
