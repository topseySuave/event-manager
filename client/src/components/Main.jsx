import React, {Component} from 'react';
import { Route, Switch} from 'react-router-dom'
import HomePage from './homepage';
import SignUp from './authentication/signup/signup';
import SignIn from './authentication/signin/signin';
import SignOut from './authentication/signout/signOut';
import CenterDetailIndex from './centerComponent/centerDetail';
import AllCentersIndex from './centerComponent/allCenters/index';
import NoMatch from './noMatch';
import createBrowserHistory from 'history/createBrowserHistory'

const history = createBrowserHistory({
    basename: '/'
});

class Main extends Component {
    render() {
        return (
            <Switch history={history}>
                <Route exact path ='/' component={HomePage} />
                <Route path ='/signup' component={SignUp} />
                <Route path = '/signin' component={SignIn} />
                <Route path = '/signout' component={SignOut} />
                <Route path = '/center/:id/:title' component={CenterDetailIndex} />
                <Route path = '/centers' component={AllCentersIndex} />
                <Route path='*' component={NoMatch} />
            </Switch>
        )
    }
}

export default Main;
