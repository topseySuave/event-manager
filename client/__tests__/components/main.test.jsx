import React from 'react';
import { mount, shallow } from 'enzyme';
import Main from
  '../../src/components/Main';

describe('Main Routing Component', () => {
  const wrapper = shallow(<Main />);
  test('have make a snapshot of the routing component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
