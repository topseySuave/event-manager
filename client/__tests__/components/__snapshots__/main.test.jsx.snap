// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Routing Component make a snapshot of the routing component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />,
        <Route
          component={[Function]}
          path="/signup"
        />,
        <Route
          component={[Function]}
          path="/signin"
        />,
        <Route
          component={[Function]}
          path="/signout"
        />,
        <Route
          component={[Function]}
          path="/center/:id/:title"
        />,
        <Route
          component={[Function]}
          path="/centers"
        />,
        <Route
          component={[Function]}
          path="/my-events"
        />,
        <Route
          component={[Function]}
          path="*"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/signup",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/signin",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/signout",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/center/:id/:title",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/centers",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/my-events",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "*",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            path="/signup"
          />,
          <Route
            component={[Function]}
            path="/signin"
          />,
          <Route
            component={[Function]}
            path="/signout"
          />,
          <Route
            component={[Function]}
            path="/center/:id/:title"
          />,
          <Route
            component={[Function]}
            path="/centers"
          />,
          <Route
            component={[Function]}
            path="/my-events"
          />,
          <Route
            component={[Function]}
            path="*"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/signup",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/signin",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/signout",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/center/:id/:title",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/centers",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/my-events",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "*",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
